autofit/optimize/non_linear/nested_sampling/dynesty.py

```diff
@@ -139,77 +139,105 @@ def _simple_fit(self, model: AbstractPriorModel, fitness_function) -> Result:
             The model which generates instances for different points in parameter space. This maps the points from unit
             cube values to physical values via the priors.
         fitness_function
-            A function that fits this model to the data, returning the likelihood of the fit.
+            A function that fits this model to the data, returning the log likelihood of the fit.
 
         Returns
         -------
-        A result object comprising the best-fit model instance, likelihood and an *Output* class that enables analysis
+        A result object comprising the best-fit model instance, log_likelihood and an *Output* class that enables analysis
         of the full chains used by the fit.
         """
-        dynesty_output = DynestyOutput(model=model, paths=self.paths)
 
         if os.path.exists("{}/{}.pickle".format(self.paths.chains_path, "dynesty")):
 
-            with open(
-                "{}/{}.pickle".format(self.paths.chains_path, "dynesty"), "rb"
-            ) as f:
-                dynesty_sampler = pickle.load(f)
+            sampler = self.load_sampler
 
         else:
 
-            dynesty_sampler = self.sampler_fom_model_and_fitness(
+            sampler = self.sampler_fom_model_and_fitness(
                 model=model, fitness_function=fitness_function
             )
 
         # These hacks are necessary to be able to pickle the sampler.
 
-        dynesty_sampler.rstate = np.random
+        sampler.rstate = np.random
         pool = Pool(processes=1)
-        dynesty_sampler.pool = pool
-        dynesty_sampler.M = pool.map
+        sampler.pool = pool
+        sampler.M = pool.map
 
         dynesty_finished = False
 
         while dynesty_finished is False:
```

Comment:
I'd accept not dynesty_finished

Alternatively you can use "while True" and "break"